syntax = "proto3";

package nvidia.inferenceserver;

//@@.. cpp:namespace:: nvidia::inferenceserver

import "grpc_service.proto";

//@@
//@@.. cpp:enum:: RequestType
//@@
//@@   Request types supported for requess.
//@@
enum RequestType {
    //@@  .. cpp:enumerator:: RequestType::INVALID = 0
    TYPE_UNKNOWN = 0;
  
    //@@  .. cpp:enumerator:: RequestType::STATUS_REQUESTOOL = 1
    TYPE_STATUS_REQUEST = 1;
  
    //@@  .. cpp:enumerator:: RequestType::INFER_REQUEST = 2
    TYPE_INFER_REQUEST = 2;
   }

//@@
//@@.. cpp:var:: message CaipRequest
//@@
//@@   Request message encapsulating multiple message types.
//@@
message CaipRequest
{
  //@@
  //@@  .. cpp:var:: RequestType request_type
  //@@
  //@@     Returns type of request enum.
  //@@
  RequestType request_type = 1;
  //@@
  //@@  .. cpp:var:: StatusReqeust status_request
  //@@
  //@@     Returns infer_request message.
  //@@
  StatusRequest status_request = 2;
  //@@
  //@@  .. cpp:var:: InferRequest infer_request
  //@@
  //@@     Returns infer_request message.
  //@@
  InferRequest infer_request = 3;
  //@@
  //@@  .. cpp:var:: string trace_message
  //@@
  //@@     Returns string-type message for diagnostics.
  //@@
  string trace_message = 4;
}
